src/mds/Migrator.cc

```diff
@@ -1114,8 +1114,12 @@ void Migrator::dispatch_export_dir(MDRequestRef& mdr, int count)
     total_exporting_size += it->second.approx_size;
 
     // start the freeze, but hold it up with an auth_pin.
-    dir->freeze_tree();
-    ceph_assert(dir->is_freezing_tree());
+    //Ephemeral distributed pinning - Only the current (child) directory should get frozen and pinned
```

Comment:
Pretty sure this isn't correct. Can you explain?

Reply:
For ephemeral distributed, you shouldn't freeze the entire tree as freezing the entire tree implies the whole subtree gets migrated. The migrator exports whatever is frozen. So here we are freezing the directory only (not the entire subtree nested below)

Reply:
I'm not sure I follow. If ephemeral distributed is set on `/home` and we're going to migrate `/home/foo`, then `/home/foo` subtree should be frozen and migrated. It sounds like you're talking about `/home` should be frozen but that wouldn't be the case?

Reply:
Great catch! Couldn't find this while debugging because there were no directories below in my test workload. i.e parent -> empty_dir so it got migrated anyways due to the `else` clause. Fixed it. Thanks!